source/common/umf_pools/disjoint_pool_config_parser.hpp

```diff
@@ -12,18 +12,36 @@
 #include <umf/pools/pool_disjoint.h>
 
 #include <memory>
+#include <mutex>
 #include <string>
+#include <thread>
 #include <unordered_map>
 
 namespace usm {
 enum DisjointPoolMemType { Host, Device, Shared, SharedReadOnly, All };
 
+typedef struct umf_disjoint_pool_config_t {
+    umf_disjoint_pool_config_t();
+
+    size_t SlabMinSize;
+    size_t MaxPoolableSize;
+    size_t Capacity;
+    size_t MinBucketSize;
+    int PoolTrace;
```

Comment:
`size_t CurPoolSize` is missing?

Reply:
CurPoolSize is only used by disjoint_pool internals (for stats), it's not needed here.